torch/utils/_traceback.py

```diff
@@ -130,21 +130,30 @@ def report_compile_source_on_error():
 
         raise exc.with_traceback(tb_next)
 
-def shorten_filename(fn):
+def shorten_filename(fn, *, base=None):
     """
     Shorten a source filepath, under the assumption that anything under torch/
     directory is "obvious" and doesn't need to be shown to user.
     """
+    if base is None:
+        base = os.path.dirname(os.path.dirname(__file__))
     # Truncate torch/foo.py to foo.py
-    prefix = os.path.commonprefix([fn, os.path.join(os.path.dirname(os.path.dirname(__file__)), "")])
-    return fn[len(prefix):]
+    try:
+        prefix = os.path.commonpath([fn, base])
```

Comment:
This is fixing a bug where 'test/test_foo.py' got shortened to 'est/test_foo.py'

Reply:
lol